### PR TITLE
SH-738: Refactoring logs 

### PR DESCRIPTION
**Pull request by aishwa8141**
Difference b/w %soureThread and %thread

> ActorLogging is asynchronous. When you log using ActorLogging, it sends a message to the logging actor, which by default runs on the default dispatcher ([thisActorSystem-akka.actor.default-dispatcher-3]). In order to print the actual thread use Mapped Diagnostic Context (MDC) that captures the Akka source(The path of the actor in which the logging was performed ), source thread( the thread in which the logging was performed) and much more in which the logging was performed.
For more info [https://stackoverflow.com/questions/60506522/thread-name-in-akka-actor-logging-regarding-pinneddispatcher](url)

### CODE REVIEW

File: group-actors/src/main/java/org/sunbird/actors/BaseActor.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public void onReceive(Object message) throws Throwable {
         onReceive(request);
         endTrace(operation);
       } catch (Exception e) {
-        logger.error("{} : message : {} {}", operation, e.getMessage(), e);
+        logger.error("{} : exception occurred: {} {}", operation, e.getMessage(), e);
```
Comment (indrajra) [inline group-actors/src/main/java/org/sunbird/actors/BaseActor.java:46]: let this log "<operation> exception" - why that semicolon which is not there in start/endTrace.

File: group-actors/src/main/java/org/sunbird/actors/CreateGroupActor.java [hunk 2]
```diff
@@ -62,17 +61,20 @@ private void createGroup(Request actorMessage) throws BaseException {
     // adding members to group, if members are provided in request
     List<Map<String, Object>> reqMemberList =
         (List<Map<String, Object>>) actorMessage.getRequest().get(JsonKey.MEMBERS);
-    logger.info("Adding members to the group: {} started", group.getName());
+    logger.info("Adding members to the group: {} groupId started", group.getName());
```
Comment (indrajra) [inline group-actors/src/main/java/org/sunbird/actors/CreateGroupActor.java:64]: Group name is not a unique thing. We could capture the name just while creating. Rest could log only group id.

File: group-actors/src/main/java/org/sunbird/service/GroupServiceImpl.java [hunk 3]
```diff
@@ -153,7 +153,7 @@ private void addActivityInfoDetails(List<Map<String, Object>> dbResActivities) {
   }
 
   /**
-   * Get all groupsIds By userIds
+   * Get all groupsIds By userIds fghjkl;
```
Comment (indrajra) [inline group-actors/src/main/java/org/sunbird/service/GroupServiceImpl.java:156]: revert this change?

File: group-actors/src/main/java/org/sunbird/service/MemberServiceImpl.java [hunk 4]
```diff
@@ -85,7 +84,7 @@ public void handleMemberOperations(Map memberOperationMap, String groupId, Strin
   public Response handleMemberAddition(
       List<Map<String, Object>> memberList, String groupId, String contextUserId)
       throws BaseException {
-    logger.info("Number of members to be added are: {}", memberList.size());
+    logger.info("Number of members to be added to the group {} are {}", groupId, memberList.size());
```
Comment (indrajra) [inline group-actors/src/main/java/org/sunbird/service/MemberServiceImpl.java:87]: do we have similar message for modified, removed?

File: service/app/utils/module/RequestIdAddFilter.java [hunk 5]
```diff
@@ -26,8 +26,7 @@ public RequestIdAddFilter(Materializer materializer) {
   public CompletionStage<Result> apply(
       Function<Http.RequestHeader, CompletionStage<Result>> nextFilter,
       Http.RequestHeader requestHeader) {
-    long startTime = System.currentTimeMillis();
-
+    MDC.clear();
```
Comment (indrajra) [inline service/app/utils/module/RequestIdAddFilter.java:29]: looks this is not the right place - pls put a comment whats the problem and how'd you think this will resolve.

Comment (aishwa8141) [reply service/app/utils/module/RequestIdAddFilter.java:29]: The problem was, sometimes it was giving previous value, I am not sure why. And one more problem I found now was, In this class sometimes it is setting reqID in MDC, sometimes it not setting. So I moved this MDC into OnRequestHandler

File: group-actors/src/main/java/org/sunbird/service/MemberServiceImpl.java (hunk 4 above)
Comment (aishwa8141) [reply group-actors/src/main/java/org/sunbird/service/MemberServiceImpl.java:87]: No, I added now

File: group-actors/src/main/java/org/sunbird/service/GroupServiceImpl.java (hunk 3 above)
Comment (aishwa8141) [reply group-actors/src/main/java/org/sunbird/service/GroupServiceImpl.java:156]: done

File: group-actors/src/main/java/org/sunbird/actors/CreateGroupActor.java (hunk 2 above)
Comment (aishwa8141) [reply group-actors/src/main/java/org/sunbird/actors/CreateGroupActor.java:64]: ok done,

File: group-actors/src/main/java/org/sunbird/service/MemberServiceImpl.java [hunk 6]
```diff
@@ -75,6 +77,10 @@ public void handleMemberOperations(Map memberOperationMap, String groupId, Strin
               .map(data -> getMemberModelForRemove(data, groupId, contextUserId))
               .collect(Collectors.toList());
       if (!removeMembers.isEmpty()) {
+        logger.info(
+            "Number of members to be modified in the group {} are {}",
```
Comment (indrajra) [inline group-actors/src/main/java/org/sunbird/service/MemberServiceImpl.java:81]: Copy paste - this needs to be removed.

File: group-actors/src/main/java/org/sunbird/actors/CreateGroupActor.java [hunk 7]
```diff
@@ -61,17 +61,15 @@ private void createGroup(Request actorMessage) throws BaseException {
     // adding members to group, if members are provided in request
     List<Map<String, Object>> reqMemberList =
         (List<Map<String, Object>>) actorMessage.getRequest().get(JsonKey.MEMBERS);
-    logger.info("Adding members to the group: {} groupId started", group.getName());
+    logger.info("Adding members to the group: {} groupId started", groupId);
```
Comment (indrajra) [inline group-actors/src/main/java/org/sunbird/actors/CreateGroupActor.java:64]: groupId started -> started.

File: group-actors/src/main/java/org/sunbird/actors/CreateGroupActor.java [hunk 8]
```diff
@@ -61,17 +61,15 @@ private void createGroup(Request actorMessage) throws BaseException {
...
       memberList.addAll(reqMemberList);
     }
     Response addMembersRes =
         memberService.handleMemberAddition(
             memberList, groupId, requestHandler.getRequestedBy(actorMessage));
     logger.info(
-        "Adding members to the group : {} ended , response {}",
-        group.getName(),
-        addMembersRes.getResult());
+        "Adding members to the group : {} ended , response {}", groupId, addMembersRes.getResult());
```
Comment (indrajra) [inline group-actors/src/main/java/org/sunbird/actors/CreateGroupActor.java:72]: getResult() - wouldn't be too big to log this?

Comment (aishwa8141) [reply group-actors/src/main/java/org/sunbird/actors/CreateGroupActor.java:72]: No, it just, a success response
